perun-core/src/main/resources/perun-core.xml

```diff
@@ -30,6 +30,7 @@ http://www.springframework.org/schema/context http://www.springframework.org/sch
 		<aop:advisor advice-ref="txAdviceNestedTransaction" pointcut="execution(* cz.metacentrum.perun.core.blImpl.MembersManagerBlImpl.validateMember(..))"/>
 		<aop:advisor advice-ref="txAdviceNestedTransaction" pointcut="execution(* cz.metacentrum.perun.core.blImpl.MembersManagerBlImpl.expireMember(..))"/>
 		<aop:advisor advice-ref="txAdviceNestedTransaction" pointcut="execution(* cz.metacentrum.perun.core.blImpl.MembersManagerBlImpl.createMemberSync(..))"/>
+		<aop:advisor advice-ref="txAdviceNestedTransaction" pointcut="execution(* cz.metacentrum.perun.core.blImpl.ResourcesManagerBlImpl.processGroupResourceActivation(..))"/>
```

Comment:
I think this will not work properly. I'm not sure but I think that the nested transaction will be finished together with the main transaction. I think we need for this a completely new transaction. However, I think we are not able to finish this during this sprint.

Reply:
Nested propagation creates a completely new transaction in this case since it is invoked from a new thread.